### DISCUSSION TITLE
테이블 캐시?

### DISCUSSION THREAD

**Discussion by CODe5753** [Well-Founded-Confidence/RealMySQL#12] [Q&A]
뭐 당연하겠지만 모든 테이블을 캐싱할 수는 없을 것 같아요.
문득 든 생각은 이 **테이블 캐시 기능이 어떻게 동작하는지**가 궁금하더라구요.
- hitrate가 높은 테이블이 캐싱될까?
- 가장 최근에 조회된 테이블이 캐싱될까?
- 테이블 자체의 크기가 많이 크다면 테이블 내에 어떤 row가 캐싱될까?
  - 인덱스 Range와 관련이 있을까?
- **결국 MySQL도 메모리를 이용해 속도를 높힐텐데, Redis와는 속도적으로 어떤 차이가 있을까?**

마지막 궁금증은 (제가 봤던 면접은 아니지만) 실제 면접에서도 나왔던 질문이었어요.
```
면접관: Redis를 왜 쓰나요?
면접자: 메모리 기반으로 속도가 빠르기 때문입니다.
면접관: 그럼 MySQL은 메모리를 안쓰나요?
면접자: ,,, 씁니다
면접관: 그럼 둘 다 메모리 기반으로 작동하는데 왜 Redis가 빠르다고 말하는 걸까요?
```

**Comment by menthamin** [comment]
MySQL 도 내부에 캐싱이 있다.

장섭님.
레디스: 모든 데이터가 메모리에 있다.
MySQL: 데이터가 메모리에도 있고 디스크에도 있다.

용수님.
레디스: 단일스레드
MySQL: 멀티스레드
- MySQL을 단일 스레드로 설정하면?

나현님.
둘다 메모리 기반
- 레디스: 인메모리(램을 사용)
- MySQL: 디스크에 저장됨, 디스크로 가려면 I/O가 발생

승현님.
원론적으로는 2개의 용도가 다르다.
MySQL에도 캐싱 기능이 있기는 하나 부족하기에 레디스로 캐싱을 한다.
램: 휘발성

그럼 MySQL은 언제 디스크 I/O 또는 메모리 접근이 일어나는가?
- 메모리

백엔드 -> 레디스
백엔드 -> DB

# Redis 이야기:
- 이슈 사례 (나현님, 승현님)

**Comment by CODe5753** [comment]
결국 이 내용은 InnoDB 버퍼풀에서 자세하게 나오는군요!
